app/controllers/api/v1/statuses_controller.rb

```diff
@@ -59,6 +59,11 @@ def destroy
     RemovalWorker.perform_async(@status.id, redraft: true)
     @status.account.statuses_count = @status.account.statuses_count - 1
 
+    status_conversation = Conversation.find(@status.conversation_id)
+    if status_conversation.status_ids.empty?
+      status_conversation.destroy
+    end
```

Comment:
This would not work, `Conversation` does not have a `status_ids`, and querying the database to find statuses with the correct conversation would be expensive, as we do not have an index on conversation.